### ISSUE TITLE
Azure OpenAI: api.env should note be required to use Azure endpoints

### ISSUE THREAD

**Issue by lbeurerkellner** [eth-sri/lmql#107]
Without api.env present, it is currently not possible to use the Azure version of OpenAI models. This should not be changed, since api.env does not actually contain any information required to access Azure models.

**Comment by hooman-bayer** [comment]
@lbeurerkellner Could you please update the documentation and post an example on how to use Azure OpenAI with lmql? None of the examples in the doc seem to be working as of today.

```python
import lmql

argmax
    "Hello[WHO]"
from
    lmql.model("openai/gpt-3.5-turbo", endpoint="azure://<ENDPOINT>", azure_api_key="<API_KEY>")
where
    STOPS_AT(WHO, "\n") and len(TOKENS(WHO)) < 10
```

Error
```
...
     44         elif line.startswith("openai-org: "):
     45             openai_org = line.split("openai-org: ")[1].strip()
---> 47 return openai_secret, openai_org

UnboundLocalError: cannot access local variable 'openai_secret' where it is not associated with a value
Output is truncated. View as a [scrollable element](command:cellOutput.enableScrolling?ef0718a3-8caa-4b86-8555-b8c2cca5f5ae) or open in a [text editor](command:workbench.action.openLargeOutput?ef0718a3-8caa-4b86-8555-b8c2cca5f5ae). Adjust cell output [settings](command:workbench.action.openSettings?%5B%22%40tag%3AnotebookOutputLayout%22%5D)...

```

Similarly setting
```
OPENAI_API_TYPE=azure
AZURE_OPENAI_<ENV_NAME>_ENDPOINT=<endpoint>
AZURE_OPENAI_<ENV_NAME>_KEY=<key>
```

Also placed the above in an `api.env` but still it does not seem to be working

**Comment by lbeurerkellner** [comment]
Thanks for raising this. We don't have Azure access ourselves, so I am basing results and experiences based on what community members in the Discord tell me about. We applied for it with the lab, and hopefully soon will have access, so we can properly test it.

Until then, I hope I can find the issue and will update the docs accordingly.

**Comment by hooman-bayer** [comment]
hey @lbeurerkellner in general lmql pipeline in using Azure OpenAI is a bit confusing. Here is how `guidance` is using it, but we really like `lmql` syntax better so it would be great to support it. I will see if we get some time to fix and do a PR no promises :) Great work anyways!

https://guidance.readthedocs.io/en/latest/example_notebooks/api_examples/llms/OpenAI.html

**Comment by lbeurerkellner** [comment]
The initial problem of this issue has been fixed in this release, so closing this for now.

I would suggest we use a new report if/when a new thing or use case comes up.

**Comment by hooman-bayer** [comment]
@lbeurerkellner just a quick feedback based on my tests, only setting the environment option works as of now:

```bash
# set the API type based on whether you want to use a completion or chat endpoint
OPENAI_API_TYPE=azure|azure-chat 

# your Azure API base URL, e.g. 'https://<YOUR_BASE>.openai.azure.com/'
OPENAI_API_BASE=<API_BASE> 

# set your API key, can also be provided per deployment 
# via OPENAI_API_KEY_{<your-deployment-name>.upper()}
OPENAI_API_KEY=<key>
```

And the following throughs the error that it does not find open_ai keys, I think its relevant to your initial comment on that `api.env` should be available
```python
my_azure_model = lmql.model(
    # the name of your deployed model/engine, e.g. 'my-model'
    "openai/<DEPLOYMENT>", 
    # set to 'azure-chat' for chat endpoints and 'azure' for completion endpoints
    api_type="azure|azure-chat",  
    # your Azure API base URL, e.g. 'https://<YOUR_BASE>.openai.azure.com/'
    api_base="<API_BASE>", 
    # your API key, can also be provided via env variable OPENAI_API_KEY 
    # or OPENAI_API_KEY_{<your-deployment-name>.upper()}
    [api_key="<API_KEY>"] , 
    # API version, defaults to '2023-05-15'
    [api_version="API_VERSION",]
    # prints the full endpoint URL to stdout on each query (alternatively OPENAI_VERBOSE=1)
    [verbose=False] 
)
```

Just for your info, anyways solution one solves this.

**Comment by lbeurerkellner** [comment]
Ah, shoot. I think I overlooked an additional api.env in another searchpath when testing. Will have a look next week. 

For now and for others, creating a dummy api.env with dummy openai.com credentials should make the second option work also.

**Comment by lbeurerkellner** [comment]
The latest 'main' should fix the issue with `api.env` now. You can try and install via:

```
pip uninstall lmql && pip install git+https://github.com/eth-sri/lmql
```

I am able to run without api.env, both via `lmql.model` and via environment variables. Let me know if that does the trick for you.

**Comment by lbeurerkellner** [comment]
This was fixed with the release of 0.0.6.6. Thanks for reporting it :)
